holoviews/plotting/bokeh/element.py

```diff
@@ -1919,6 +1928,30 @@ def initialize_plot(self, ranges=None, plot=None, plots=None, source=None):
 
         return plot
 
+    def _apply_hard_bounds(self, element, ranges):
+        """
+        Apply hard bounds to the x and y ranges of the plot. If xlim/ylim is set, limit the
+        initial viewable range to xlim/ylim, but allow navigation up to the abs max between
+        the data range and xlim/ylim. If dim range is set (e.g. via redim.range), enforce
+        as hard bounds.
+
+        """
+
+        def validate_bound(bound):
+            return bound if util.isfinite(bound) else None
+
+        min_extent_x, min_extent_y, max_extent_x, max_extent_y = map(validate_bound, self.get_extents(element, ranges, range_type='combined', lims_as_soft_ranges=True))
```

Comment:
```suggestion
        min_extent_x, min_extent_y, max_extent_x, max_extent_y = map(
            validate_bound, self.get_extents(element, ranges, range_type='combined', lims_as_soft_ranges=True)
        )
```
Nit. The line is pretty long.